google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/appengine/util/AppEngineUtilTest.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2016 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.util;
+
+import static org.junit.Assert.assertEquals;
+
+import org.joda.time.DateTimeUtils;
+import org.junit.After;
+import org.junit.Test;
+
+/**
+ * Tests App Engine Utilities
+ * Uses an external trusted datetime to milli converter to set the datetime to the desired value.
+ */
+public class AppEngineUtilTest {
+
+  @Test
+  public void testDateTimeToVersion() {
```

Comment:
Could you explain a bit? If I use a clock internally with real time then the test feels a bit flaky to me because it depends on when the test is run. If the clock is set to a preconfigured datetime then the problem is circular because it would require using a formatter to set it (which is what we are testing in the first place). I tried mocking DateTime but didn't have much luck with that approach.